File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/enterprise/address/ui/EnterpriseAddressViewComponent.java
package com.fintechplatform.ui.enterprise.address.ui;

import com.fintechplatform.api.enterprise.api.EnterpriseAPIModule;
import com.fintechplatform.api.log.LogModule;
import com.fintechplatform.api.net.NetModule;
import com.fintechplatform.ui.alert.AlertHelpersModule;
import com.fintechplatform.ui.enterprise.db.enterprise.EnterprisePersistanceDBModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component (modules = {
        EnterpriseAddressPresenterModule.class,
        EnterpriseAPIModule.class,
        NetModule.class,
        LogModule.class,
        AlertHelpersModule.class,
        EnterprisePersistanceDBModule.class
})
public interface EnterpriseAddressViewComponent {
    void inject(EnterpriseAddressActivity activity);
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/enterprise/address/di/EnterpriseAddressViewComponent.kt
package com.fintechplatform.ui.enterprise.address.di

import com.fintechplatform.api.enterprise.api.EnterpriseAPIModule
import com.fintechplatform.api.log.LogModule
import com.fintechplatform.api.net.NetModule
import com.fintechplatform.api.profile.api.IdsDocumentsAPIModule
import com.fintechplatform.ui.alert.AlertHelpersModule
import com.fintechplatform.ui.enterprise.address.EnterpriseAddressFragment
import com.fintechplatform.ui.enterprise.db.enterprise.EnterprisePersistanceDBModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [EnterpriseAddressPresenterModule::class, EnterpriseAPIModule::class, IdsDocumentsAPIModule::class, NetModule::class, LogModule::class, AlertHelpersModule::class, EnterprisePersistanceDBModule::class])
interface EnterpriseAddressViewComponent {
    fun inject(fragment: EnterpriseAddressFragment)
}
